jsettlers.common/src/main/java/jsettlers/common/map/shapes/MapRectangle.java

```diff
@@ -148,17 +145,17 @@ public boolean iterate(IBooleanCoordinateFunction function) {
 
 		@Override
 		public boolean hasNext() {
-			return relativeY < height && width > 0;
+			return relativeY <= height && relativeX <= width;
 		}
 
 		@Override
 		public ShortPoint2D next() {
-			if (relativeY < height && width > 0) {
-				int x = getLineStartX(relativeY) + relativeX;
-				int y = getLineY(relativeY);
+			if (hasNext()) {
+				int x = minX + relativeX;
```

Comment:
Have a look at getLineStartX(), it's a bit more complicated than that ;-)

Reply:
But the existing MapAreasTest did not fail with this implementation. Maybe it does not cover all the edge cases.